### DISCUSSION TITLE
Aggregating data from multiple Garmin Connect a/c for research?

### DISCUSSION THREAD

**Discussion by tschuler** [tcgoetz/GarminDB#112] [General]
Hi @tcgoetz and GarminDB community,

Very nice project, many thanks. Currently running off the dev branch with latest fix for issue re the header.

I am a cancer doctor and I am use Garmin vivosmart 4s for a pilot research study (n=30) with constraint budget. Looking into using GarminDB instead of https://www.fitabase.com/ to pool data from several accounts. 

Maybe its naive but was thinking along the lines of:
- creating several sqlite DBs for each a/c by running GarminDB make script several times for each a/c via a nightly cron script
- would require to specify destination directory via config file (unless I am mistaken currently always gos to ~/HealthData)
- querying across the separate sqlite DBs for data analysis as per: https://stackoverflow.com/questions/6824717/sqlite-how-do-you-join-tables-from-different-databases

Has this been done before? Maybe is there even a way to specify a destination directory already? Other ideas re this requirement?

Many thanks for any tips/pointers etc.

Best, Thilo

**Comment by tcgoetz** [comment]
> Hi @tcgoetz and GarminDB community,
> 
> Very nice project, many thanks. Currently running off the dev branch with latest fix for issue re the header.
> 
> I am a cancer doctor and I am use Garmin vivosmart 4s for a pilot research study (n=30) with constraint budget. Looking into using GarminDB instead of https://www.fitabase.com/ to pool data from several accounts.
> 
> Maybe its naive but was thinking along the lines of:
> 
>     * creating several sqlite DBs for each a/c by running GarminDB make script several times for each a/c via a nightly cron script
> 
>     * would require to specify destination directory via config file (unless I am mistaken currently always gos to ~/HealthData)

Location of the DB directory is controller here: https://github.com/tcgoetz/GarminDB/blob/e672b4d69a221854ab96690361e371bc21d796ca/garmin_db_config.py#L15

> 
>     * querying across the separate sqlite DBs for data analysis as per: https://stackoverflow.com/questions/6824717/sqlite-how-do-you-join-tables-from-different-databases
> 

I'm pretty sure there are examples of querying across DB files in the codebase since there are multiple DB files in use.

> 
> Has this been done before? Maybe is there even a way to specify a destination directory already? Other ideas re this requirement?
> 
> Many thanks for any tips/pointers etc.
> 
What you describe should work. The process can probably be streamlined by passing the config info on the command line or similar.

Let me know if your have more questions.

> Best, Thilo

**Reply by tschuler** [reply]
Thanks for your quick reply. 

Great idea to pass 'base_dir' optionally via command line. Would adding this snippet in [Makefile](https://github.com/tcgoetz/GarminDB/blob/master/Makefile) do it? Sorry I have limited programming skills.
```
ifdef base_dir
  HEALTH_DATA_DIR=$(base_dir)
else
  HEALTH_DATA_DIR=$(shell $(PYTHON) -c 'from garmin_db_config_manager import GarminDBConfigManager; print(GarminDBConfigManager.get_base_dir())')
endif
```
Could you pls point to one of the cross DB queries in your code (searched for 'attach' but must be doing it differently...

Many thanks again, Thilo

**Reply by skrasnow** [reply]
FYI - attached is a SQLite query across multiple databases.

Stan 
[SJK Double Join Weekly Activity Steps & Total Steps & Activity Distance.txt](https://github.com/tcgoetz/GarminDB/files/6068812/SJK.Double.Join.Weekly.Activity.Steps.Total.Steps.Activity.Distance.txt)

**Reply by tcgoetz** [reply]
> Great idea to pass 'base_dir' optionally via command line. Would adding this snippet in [Makefile](https://github.com/tcgoetz/GarminDB/blob/master/Makefile) do it? Sorry I have limited programming skills.

That would only effect the Makefile and not effect garmin.py. There isn't current plumbing to change the value for a running garmin.py. I've been thinking about that in the background, but haven't had any time to do anything about it yet.

**Reply by tschuler** [reply]
@skrasnow Thanks for sharing

@tcgoetz Thanks for considering:
- are you thinking of a "setter" classmethod in [garmin_db_config_manager.py](https://github.com/tcgoetz/GarminDB/blob/master/garmin_db_config_manager.py) that can set the base_dir at run-time? 
- Sorry have never coded in Python, don't understand where the 'cls' object gets instantiated? 
- Let me know if you think you will get to this in future with rough ETA, if you don't have availability, I can try to tinker...

**Reply by tschuler** [reply]
@tcgoetz Sorry for hassling. Any thoughts on my last post? Many thanks, Thilo (thilo.schuler@gmail.com)

**Reply by joelgerard** [reply]
Did you get this working? I did this by resetting the HOME directory and just aggregating it all downstream.
